network-api/networkapi/wagtailpages/pagemodels/base.py

```diff
@@ -33,6 +36,37 @@ class BasePage(FoundationMetadataPageMixin, FoundationNavigationPageMixin, Page)
     class Meta:
         abstract = True
 
+    def get_donate_banner(self, request):
+        # Check if theres an active A/B test for the homepage.
+        homepage = self.get_ancestors().type(Homepage).specific().first()
+        active_ab_test = AbTest.objects.filter(page=homepage, status=AbTest.STATUS_RUNNING).first()
+
+        # If theres no A/B test found, return the donate_banner field as usual.
+        if not active_ab_test:
+            return homepage.donate_banner
+
+        # Check for the cookie specific to this A/B test.
+        test_cookie_name = f"wagtail-ab-testing_{active_ab_test.id}_version"
+        test_version = request.COOKIES.get(test_cookie_name)
+
+        # If no cookie is found, add user as a participant in test, and set a cookie for their assigned version.
+        if not test_version:
+            test_version = active_ab_test.add_participant()
```

Comment:
The [add_participant()](https://github.com/wagtail-nest/wagtail-ab-testing/blob/main/wagtail_ab_testing/models.py#L285) method adds the user as a participant to the A/B test, and returns their assigned version of either "control" or "variant".

Reply:
Hi @mmmavis!

If we honor the user's Do Not Track preference, we will show them the "control" version of the donate banner.

The "control" version refers to whatever was set in the `donate_banner` field before a different banner was selected and the "Save and create AB test" option was clicked.